src/createFormValues.js

```diff
@@ -25,57 +27,65 @@ const createValues = ({ getIn }: Structure<*, *>): FormValuesInterface => (
             'formValues() must be used inside a React tree decorated with reduxForm()'
           )
         }
-        let valuesMap: PropPath[]
+        this.updateComponent(props)
+      }
+      componentWillReceiveProps(props) {
+        if (typeof firstArg === 'function') {
+          this.updateComponent(props)
+        }
+      }
+      render() {
+        const { Component } = this
+        return (
+          <Component
+            // so that the connected component updates props when sectionPrefix has changed
+            sectionPrefix={this.context._reduxForm.sectionPrefix}
+            {...this.props}
+          />
+        )
+      }
+      updateComponent(props) {
+        let valuesMap: Object
         const resolvedFirstArg: string | Object =
           typeof firstArg === 'function' ? firstArg(props) : firstArg
         if (typeof resolvedFirstArg === 'string') {
-          valuesMap = [
-            resolvedFirstArg,
-            ...rest
-          ].map((k: string): PropPath => ({
-            prop: k,
-            path: k
-          }))
+          valuesMap = {}
```

Comment:
you better use reduce function over init empty object and then go over the array with foreach to init the values

[resolvedFirstArg, ...rest].reduce((result, current) => { ...result, [current]: current }, {});

Reply:
Fixed